core/src/main/java/org/elasticsearch/index/query/IndexQueryParserService.java

```diff
@@ -339,4 +349,9 @@ public boolean matchesIndices(String... indices) {
         }
         return false;
     }
+
+    public List<Object> handleTermsLookup(TermsLookup termsLookup) {
+        termsLookup.setIndex((termsLookup.getIndex() == null) ? this.index().name() : termsLookup.getIndex());
```

Comment:
I think it's a bit odd to mutate the terms lookup object once passed int, I would prefer if this logic was oustide of this method. Can we move it to toQuery and set the index there?